File: PersonaFlowReader/src/dataAccess/dataTypes/InnerFileAddress.java
package dataAccess.dataTypes;

import dataAccess.FileReadWriteUtils;

import java.io.IOException;
import java.io.RandomAccessFile;
import java.nio.ByteOrder;

public class InnerFileAddress {
    public final int startAddr;
    public final int endAddr;
    public final long currAddr;

    public InnerFileAddress(int startAddr, int endAddr, long currAddr) {
        this.startAddr = startAddr;
        this.endAddr = endAddr;
        this.currAddr = currAddr;
    }

    public static InnerFileAddress getFileAddress(RandomAccessFile file, ByteOrder order) throws IOException {
        long currAddr = file.getFilePointer();
        int startAddr = FileReadWriteUtils.readInt(file, order);
        int endAddr = FileReadWriteUtils.readInt(file, order);
        return new InnerFileAddress(startAddr, endAddr, currAddr);
    }
}

File: PersonaFlowReader/src/dataAccess/dataTypes/InnerFileAddress.kt
package dataAccess.dataTypes

import dataAccess.readInt
import java.io.IOException
import java.io.RandomAccessFile
import java.nio.ByteOrder

data class InnerFileAddress(
    val startAddr: Int,
    val endAddr: Int,
    val currAddr: Long
)

@Throws(IOException::class)
fun getFileAddress(file: RandomAccessFile, order: ByteOrder): InnerFileAddress {
    val currAddr = file.filePointer
    val startAddr = readInt(file, order)
    val endAddr = readInt(file, order)
    return InnerFileAddress(startAddr, endAddr, currAddr)
}